Fix CMakeLists + package.xmls

mintar opened the pull request:
This fixes compilation errors when rosjava is installed on the system.
It also removes a lot of errors reported by catkin_lint.

Fixes #537 .

mintar committed:
Fix CMakeLists + package.xmls

This fixes compilation errors when rosjava is installed on the system.
It also removes a lot of errors reported by catkin_lint.

Fixes #537 .

spmaniato commented:
@mintar Were there no complaints about `move_base` ? See [this comment](https://github.com/ros-planning/navigation/issues/483#issuecomment-235446009) in #483

mintar commented:
@spmaniato I didn't even run `catkin_lint` on `move_base`, since with this PR it compiles for me with `rosjava` installed. But one should definitely go through all the packages again and fix all `catkin_lint` errors.

Some more remarks (they may be obvious, but nevertheless):

* The bug #537 only occurs when `rosjava` is installed *and* `navigation` is checked out as a source package into the Catkin workspace you're building.
* Even after applying this PR, I initially need to run `catkin_make` 2-3 times before it eventually succeeds due to some problems with `dynamic_reconfigure`. (Without this PR, it doesn't build at all.)

mintar commented:
Ping

caicaixia commented:
Hello,  these changes haven't released yet, right? I just upgrade the ros-indigo-navigation, but the version is: 1.12.13-0trusty-20161128-101040-0800.  And still have ros java problem :(

mintar commented:
@caicaixia : No, I'm still waiting for this PR to be merged. But this bug only concerns source installs anyway, so it doesn't matter if this is released or not.

DLu commented on costmap_2d/CMakeLists.txt:
```diff
@@ -60,8 +61,6 @@ generate_dynamic_reconfigure_options(
     INCLUDE_DIRS
         include
-        ${EIGEN_INCLUDE_DIRS}
-        ${PCL_INCLUDE_DIRS}
```
Why are these removed?

mintar replied:
Because there already is a catkin_package(DEPENDS PCL Eigen). This means that PCL + Eigen are already added as dependencies for any project that depends on `costmap_2d`, and it's Eigen's (resp. PCL's) job to export their own include dirs. The `catkin_package(INCLUDE_DIRS)` should only export our own include dirs.

Removing these two lines makes the following `catkin_lint` error go away:
```
costmap_2d: CMakeLists.txt(61): error: catkin_package() exports non-package include path
     * You listed one or more include paths in the INCLUDE_DIRS stanza
     * of your catkin_package() call which are not part of your
     * package. If you want to export include paths of other modules,
     * use find_package(), find_path(), and/or find_library() and add
     * the dependency to the DEPENDS stanza.
```

DLu reviewed (commented):
Overall, looks good. I merged things in the wrong order (see #549), so could you please merge the latest indigo-devel?

mintar commented:
I've rebased this branch onto the latest indigo-devel, so it should be good to merge.

mintar committed:
Fix more CMakeLists.txt + package.xmls

mintar committed:
amcl: disambiguate include path

This fixes the following catkin_lint warning:

    amcl: warning: include paths 'include/amcl/map' and 'include' are ambiguous
         * You have used two include paths where one is a parent of the
         * other. Thus the same headers can be included with two different
         * include paths which may confuse users. It is recommended that
         * you keep your include paths consistent.

mintar commented:
I've just added two more commits that fix 87 out of 90 `catkin_lint` errors (the rest are false positives).

mikeferguson commented:
@mintar -- could you rebase this on top of latest changes (there is a conflict) and then we can get this merged. Thanks!

mintar commented:
@mikeferguson Ok, rebased!

mintar commented:
ping

mikeferguson commented:
Test failures are unrelated. I'm going to squash-merge to make it easier to forward port to K+L

mintar commented:
Hooray! :-)
